### PR TITLE
Implemented backward compatibility tests for `JournalStorage`

### PR DESCRIPTION
**Pull request by porink0424**
## Motivation

In preparation for the official release of `JournalStorage`, this PR added tests to verify that log files of `JournalStorage` have backward compatibility.

## Description of the changes

- Implemented the code for creating a sample log file
- Added a sample log
- Implemented backward compatibility tests

### CODE REVIEW

**Comment by nabenabe0928** [comment]
@porink0424 Thank you for the PR!

@eukaryo @c-bata 
Could you review this PR?

File: tests/storages_tests/journal_tests/create_journal.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    parser.add_argument(
+        "--storage-url",
+        default=f"{os.path.dirname(__file__)}/assets/{optuna.__version__}.log",
+    )
+    args = parser.parse_args()
+
+    storage = optuna.storages.JournalStorage(optuna.storages.JournalFileStorage(args.storage_url))
+
+    # Empty study
+    optuna.create_study(storage=storage, study_name="single_empty")
```
Comment (c-bata) [inline tests/storages_tests/journal_tests/create_journal.py:22]: [nit] Could you add a test case to check `single_empty` study is a truly empty like below?

```python
study = optuna.load_study(...)

assert study.directions == ["minimize"]
assert len(study.user_attrs) == 0
assert len(study.trials) == 0
```

File: tests/storages_tests/journal_tests/assets/4.0.0.dev.log [hunk 2]
```diff
@@ -0,0 +1,92 @@
+{"op_code": 0, "worker_id": "a4572293-4260-45be-9d2a-489e532dc113-140704507377600", "study_name": "single_empty", "directions": [1]}
```
Comment (c-bata) [inline tests/storages_tests/journal_tests/assets/4.0.0.dev.log:1]: Just for the record, I confirmed that the asset log file contains [all kinds of `op_type`](https://github.com/optuna/optuna/blob/6aa86fc37811a421d444b6426eae1fc7c544ea96/optuna/storages/_journal/storage.py#L39-L49) 👍 

```python
>>> import json
>>> with open("tests/storages_tests/journal_tests/assets/4.0.0.dev.log") as f:
...     lines = f.readlines()
...
>>> d = [json.loads(l) for l in lines]
>>> set([a["op_code"] for a in d])
{0, 1, 2, 3, 4, 5, 6, 7, 8, 9}
```

**Review by c-bata — COMMENTED** [review]
Looks almost good to me.

Like `rdb_tests` directory, how about moving the following file under the `journal_tests` directory?
https://github.com/optuna/optuna/blob/master/tests/storages_tests/test_journal.py

**Comment by porink0424** [comment]
@c-bata 
Thank you for your review. I have updated the code following your reviews!

File: tests/storages_tests/journal_tests/test_storage.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
+import os
```
Comment (c-bata) [inline tests/storages_tests/journal_tests/test_storage.py:1]: [nit] Might be a little bit better to rename this into `test_log_compatibility.py`.

**Review by c-bata — APPROVED** [review]
LGTM with a nit!

**Review by c-bata — APPROVED** [review]

**Review by eukaryo — APPROVED** [review]
LGTM. I reviewed the code and results of tests. Thank you for the solid work.
